app/assets/js/rtm-upload-terms.js

```diff
@@ -1,139 +1,163 @@
-			terms_conditions_checkbox.removeAttr( 'checked' );
-			jQuery( '.rt_alert_msg' ).remove();
-		}
-		return true;
-	});
-
-	rtMediaHook.register( 'rtmedia_js_after_activity_added', function() {
-		var rtmedia_terms_conditions = $( '#rtmedia_upload_terms_conditions' );
-		if ( rtmedia_terms_conditions && rtmedia_terms_conditions.is(':checked') ) {
-			rtmedia_terms_conditions.prop( 'checked', false );
-		}
-	} );
+    /**
+     * Check for the terms checkbox to be checked before the media is uploaded.
+     * Show warning if unchecked else proceed.
+     */
+    rtMediaHook.register( 'rtmedia_js_upload_file', function ( args ) {
+        if ( false === args ) {
+
+            return args;
+        }
+
+        /**
+         * Show the warning message if the terms checkbox is enabled in the admin panel.
+         * Extra check if the HTML not exists, or has been removed intentionally.
+         *
+         * @author Adarsh Verma <adarsh.verma@rtcamp.com>
+         */
+        var activity_terms_enabled = rtmedia_upload_terms_data.activity_terms_enabled;
+        if ( 'false' === activity_terms_enabled ) {
+
+            return false;
+        }
+
+        var terms_conditions_checkbox = jQuery( '#rtmedia_upload_terms_conditions' );
+        if ( 0 === terms_conditions_checkbox.length ) {
+            rtp_display_terms_warning( jQuery( '#whats-new-options' ), rtmedia_upload_terms_data.message );
+
+            return false;
+        } else if ( 1 === terms_conditions_checkbox.length && ! terms_conditions_checkbox.is(':checked') ) {
+            rtp_display_terms_warning( terms_conditions_checkbox.parent( '.rtmedia-upload-terms' ), rtmedia_upload_terms_data.message );
+
+            return false;
+        } else {
+
+            return true;
+        }
+
+    } );
+
+    /**
+     * Show the warning message if the terms checkbox is unchecked before posting the media update.
+     */
+    rtMediaHook.register( 'rtmedia_js_before_activity_added', function ( args ) {
+        var terms_conditions_checkbox, form;
+        var whats_new_submit = jQuery( '#aw-whats-new-submit' );
+
+        if ( args && 'activity' === args.src ) {
+            form = jQuery( '#whats-new-form' );
+            terms_conditions_checkbox = form.find( '#rtmedia_upload_terms_conditions' );
+        } else {
+            terms_conditions_checkbox = jQuery( '#rtmedia_upload_terms_conditions' );
+        }
+
+        if ( 1 === terms_conditions_checkbox.length ) {
+            terms_conditions_checkbox.removeAttr( 'disabled' );
+
+            if ( false === args ) {
+                whats_new_submit.removeAttr( 'disabled' );
+                whats_new_submit.removeClass( 'loading' );
+
+                return args;
+            }
+
+            if ( ! terms_conditions_checkbox.is( ':checked' ) ) {
+                whats_new_submit.removeAttr( 'disabled' );
+                whats_new_submit.removeClass( 'loading' );
+                if ( args && 'activity' === args.src ) {
```

Comment:
Leave blank line above.

Reply:
@SID177 there is already a blank space before `if` statement.